lib/helpers.js

```diff
@@ -13,21 +27,63 @@ export function locToRange(loc: Object): [[number, number], [number, number]] {
 
 export function processDeclaration(entries: Array<Object>, sourceFile: string): Array<Object> {
   const toReturn = []
+  
+  const openProjects = atom.project.getPaths()
+  let projectPath
+  for (let i = 0, length = openProjects.length; i < length; ++i) {
+    if (sourceFile.indexOf(openProjects[i]) === 0) {
+      projectPath = openProjects[i]
+      break
+    }
+  }
+  
   for (let i = 0, length = entries.length; i < length; ++i) {
     const entry = entries[i]
     let filePath = entry.source.filePath
+
     if (filePath) {
       if (resolve.isCore(filePath)) {
         continue
       }
+    
+      let hasError = false
+    
       try {
         filePath = resolve.sync(filePath, {
           basedir: Path.dirname(sourceFile),
         })
       } catch (_) {
+        hasError = true
+        
+        for (let j = 0, rootLength = rootsPath.length; j < rootLength; ++j) {
+          hasError = false
+          try {
+            let bPath = Path.join(projectPath, rootsPath[j], filePath);
+            let fileName = '';
+            filesExt.forEach(function (ext) {
+              if (fs.existsSync(`${bPath}${ext}`)) {
+                bPath += ext
+                fileName = bPath.replace(/^.*[\\\/]/, '')
+                bPath = Path.dirname(bPath)
+              }
+            });
+        
+            filePath = resolve.sync(`./${fileName}`, {
```

Comment:
We do not want to use any sync operations because this code is executed on the same process as the Atom UI so anything sync will jam the UI, can you replace it with async resolve?